x/programs/rust/examples/counter/src/lib.rs

```diff
@@ -284,5 +237,66 @@ mod tests {
                 .filter_map(|resp| resp.error.as_ref())
                 .next()
         );
+
+        let value = bytes_to_u64(
+            simulator
+                .run_step(
```

Comment:
Can we not make `run_step`'s return value generic? Something is super off about the way this works. I think it's that we're dealing with Go responses that don't have `enums`, but you can just write a custom deserializer such that you don't need `serde_as` (dependency dropped is a win), and ignore whatever is in the `response` field if the `error` field is populated.

Reply:
Not using `serde_as` for the deserializing of data from Go won't help dropping the dependency because it's also used in the `Param` enum. it looks more compact that way but would you prefer writing a custom deserializer method to do that by hand? I also made the `run_step` return data generic.